### PR TITLE
all: refine error process

### PR DESCRIPTION
**Pull request by chai2010**

### CODE REVIEW

File: pkg/cmd/runtime/main.go [hunk 1]
```diff
@@ -33,14 +33,16 @@ func Main(cfg *config.Config) {
 
 	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", cfg.Runtime.Port))
 	if err != nil {
-		log.Fatalf("failed to listen: %v", err)
+		err = errors.WithStack(err)
+		logger.Fatalf("failed to listen: %v", err)
```
Comment (chilianyi) [inline pkg/cmd/runtime/main.go]: What is the different between %+v and %v?

Comment (chai2010) [reply pkg/cmd/runtime/main.go]: `%+v` is the extended format. Each Frame of the error's StackTrace will be printed in detail.

https://godoc.org/github.com/pkg/errors

Comment (chilianyi) [reply pkg/cmd/runtime/main.go]: got it, the code here should to be +v or not?

Comment (chai2010) [reply pkg/cmd/runtime/main.go]: fixed

**Comment by chai2010** [comment]
ping

File: pkg/cmd/app/main.go [hunk 2]
```diff
@@ -53,7 +55,8 @@ type AppServer struct {
 func NewAppServer(cfg *config.Database) *AppServer {
 	db, err := db.OpenAppDatabase(cfg)
 	if err != nil {
-		log.Fatal(err)
+		err = errors.WithStack(err)
+		logger.Fatal(err)
```
Comment (chilianyi) [inline pkg/cmd/app/main.go]: Whether it should be Fatalf("%+v", err)

Comment (chai2010) [reply pkg/cmd/app/main.go]: fixed

File: pkg/cmd/api/main.go [hunk 3]
```diff
@@ -39,7 +40,7 @@ func Main(cfg *config.Config) {
 	logger.Printf("Api service start http://%s:%d\n", cfg.Api.Host, cfg.Api.Port)
 
 	if err := run(cfg); err != nil {
-		log.Fatal(err)
+		log.Fatalf("%+v", err)
```
Comment (chilianyi) [inline pkg/cmd/api/main.go]: Whether it should be logger?

Comment (chai2010) [reply pkg/cmd/api/main.go]: yes, we should use logger instead of std log

Comment (chilianyi) [reply pkg/cmd/api/main.go]: if all logger.Fatalf should be like this, can we change looger.Fatalf to 
```
func Fatalf(format string, args ...interface{}) {
	if len(args) > 0 {
		args[0] = errors.WithStack(args[0])
	}
	logger.Fatalf(format, args...)
	// Make sure fatal logs will exit.
	os.Exit(1)
}
```

Comment (chai2010) [reply pkg/cmd/api/main.go]: it is a wrong practice, `errors.WithStack` will include the `logger.Fatalf` call frames.

Comment (chilianyi) [reply pkg/cmd/api/main.go]: ok, got it

**Pull request merged**